card/lib/card/query/card_query/match_attributes.rb

```diff
@@ -0,0 +1,63 @@
+class Card
+  module Query
+    class CardQuery
+      # handle special CQL attributes
+      module MatchAttributes
+        def found_by val
+          found_by_cards(val).compact.each do |card|
+            subquery found_by_statement(card).merge(fasten: :direct, context: card.name)
+          end
+        end
+
+        # Implements the match attribute that matches always against content and name.
+        # Currently that's different from the match operator that can be restricted to
+        # names or content.
+        # Example: { match: "name or content" } vs. { name: ["match", "a name"] }
+        # TODO: unify handling for both using full text indexing
```

Comment:
this TODO is the most important comment.  MySQL full text indexing has come a long way since we started working on this.  I'm hoping we can build a general solution for postgres and mysql and use that in all our matching.